src/main/java/com/jagrosh/giveawaybot/commands/SettingsCommand.java

```diff
@@ -47,12 +56,94 @@ public SettingsCommand(Bot bot)
     @Override
     protected void execute(CommandEvent event)
     {
-        EmbedBuilder eb = new EmbedBuilder();
         GuildSettings settings = bot.getDatabase().settings.getSettings(event.getGuild().getIdLong());
-        PremiumLevel premium = bot.getDatabase().premium.getPremiumLevel(event.getGuild());
-        
+        PremiumLevel level = bot.getDatabase().premium.getPremiumLevel(event.getGuild());
+
+        String[] args = event.getArgs().split("\\s+", 2);
+
+        switch (args[0])
+        {
+            case "emoji":
+                emojiBlock(event, level, args);
+                break;
+            case "clear":
+            case "reset":
+                resetBlock(event);
+                break;
+            default:
+                defaultBlock(event, new EmbedBuilder(), settings, level);
+        }
```

Comment:
This shouldn't be done with a switch statement, this should just use children commands

Reply:
as in `EmojiCommand extends SettingsCommand` and registering it?